mongoid.js

```diff
@@ -109,7 +109,12 @@ MongoId.prototype.mongoid = MongoId.prototype.fetch;
 var _zeroPadding = ["", "0", "00", "000", "0000", "00000", "000000", "0000000"];
 function hexFormat(n, width) {
     var s = n.toString(16);
-    return _zeroPadding[width - s.length] + s;
+    var paddingNeeded = width - s.length;
+    if (paddingNeeded > 0) {
```

Comment:
this could be simpler written as `if (s.length <= width)`

Reply:
or even simpler if the incoming process id is masked first, ie
`this.processIdStr = hexFormat(machineId, 6) + hexFormat(processId & 0xFFFF, 4);`
(the other formatted fields, timestamp, sequence and machineId, are already length checked)